### DISCUSSION TITLE
[venus-dev] 统一各个仓库的PR模板

### DISCUSSION THREAD

**Discussion by LinZexiao** [filecoin-project/venus#5155] [Ideas]
目前我们项目只有venus 提供了PR模板
可以参考 [venus PR 模板](https://github.com/filecoin-project/venus/blob/master/.github/pull_request_template.md)

**Comment by LinZexiao** [comment]
cluster 也提供了类似的模板：
中英双语言，去除了 标题的要求

[venus cluster PR 模板](https://github.com/ipfs-force-community/venus-cluster/blob/main/.github/pull_request_template.md)

**Comment by diwufeiwen** [comment]
缺少pr模板的组件：

venus-messager，venus-auth，venus-wallet，venus-market，venus-gateway

venus的那个pr是不是契合我们的需求，需要评估。


缺少dependabot配置的组件：

venus-messager，venus-auth，venus-wallet，venus-market，venus-gateway

**Comment by LinZexiao** [comment]
添加或修改以下仓库的PR模板：
- [x]  Venus https://github.com/filecoin-project/venus/pull/5123
- [x] auth https://github.com/filecoin-project/venus-auth/pull/91
- [x] gateway https://github.com/ipfs-force-community/venus-gateway/pull/70
- [x] miner https://github.com/filecoin-project/venus-miner/pull/121
- [x] wallet https://github.com/filecoin-project/venus-wallet/pull/94
- [x] messager https://github.com/filecoin-project/venus-messager/pull/228
- [x] market https://github.com/filecoin-project/venus-market/pull/174
- [x] docs https://github.com/filecoin-project/venus-docs/pull/337
- [ ] cluster https://github.com/ipfs-force-community/venus-cluster/pull/367

**Comment by LinZexiao** [comment]
先合并几个常用的仓库使用一段时间，然后推广到其他所有的仓库

**Comment by simlecode** [comment]
> > ## 自查清单 (Checklist)
> > 在你认为本 PR 满足被审阅的标准之前，需要确保 / Before you mark the PR ready for review, please make sure that:
> > 
> > * [ ]  符合Venus项目管理规范中关于PR的[相关标准](https://github.com/ipfs-force-community/dev-guidances/blob/master/%E9%A1%B9%E7%9B%AE%E7%AE%A1%E7%90%86/Venus/PR%E5%91%BD%E5%90%8D%E8%A7%84%E8%8C%83.md) / The PR follows the PR standards set out in the Venus project management guidelines
> > * [ ]  具有清晰明确的 commit message / All commits have a clear commit message.
> >   修改后的 PR 模版中的自查清单
> > * [ ]  包含必要的测试用例 / This PR has tests for new functionality or change in behaviour
> > * [ ]  包含必要的指南或文档 / If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included
> > * [ ]  通过必要的检查项 / All checks are green
> 
> 其中 "包含必要的测试用例", "包含必要的指南或文档" 从语义上来说是本次 PR 包含测试用例和文档才需要勾选，但我觉得这两个改成检查项 "包含必要的测试用例或不需要", "包含必要的指南或文档或不需要"，可能更加符合 “自查清单“。
> 
> 参考案例: [ant-design PULL_REQUEST_TEMPLATE CN](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md#%EF%B8%8F-%E8%AF%B7%E6%B1%82%E5%90%88%E5%B9%B6%E5%89%8D%E7%9A%84%E8%87%AA%E6%9F%A5%E6%B8%85%E5%8D%95), [ant-design PULL_REQUEST_TEMPLATE EN](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE.md#%EF%B8%8F-self-check-before-merge)

**Comment by simlecode** [comment]
## 现 pr 模板中自查清单功能讨论

venus中对于pr模板中的讨论：https://github.com/filecoin-project/venus/pull/5123

现自查清单包含内容：

![image](https://user-images.githubusercontent.com/69969590/182751663-48dea963-692e-4208-aaf4-4e84ddab531e.png)


观点一：自查清单主要是pr提交者用于自查，有其它方法向review的人传递信息

观点二：自查清单关于测试及文档的描述不能准确向review者提供参考信息

**Comment by 0x5459** [comment]
# PR 模版中是否还需要 `附注 (Additional Info)`?

> 附注 这一块在我们的实践中，基本没有用到。
> 别的项目有使用这一块的较好实践么？
> 因为这个模板本身也是从 venus 项目中借鉴过来的

https://github.com/ipfs-force-community/venus-cluster/pull/367#pullrequestreview-1064529316

@diwufeiwen @LinZexiao @simlecode

**Reply by 0x5459** [reply]
我个人觉得在我们的实践中基本没有用到。如果有附注可以直接写在 `改动`中，所以可以移除

**Reply by LinZexiao** [reply]
目前我以及我见到过的PR都没有使用过这个区域，不过我个人认为，它可以保留，因为附注信息本来也就是指额外的一些信息，所以不常用到是符合预期的。此外，我觉得确实也需要一个区域用来记录除了变动和相关issue之外的信息，传递给 review 和 分支发布 的执行者。我觉得这一部分可以在以后分支发布等的一些规范建立起来之后，在相应的工程实践中，起到一定的作用

**Reply by LinZexiao** [reply]
@zl03jsj @hunjixin

**Reply by Fatman13** [reply]
对，确实用到的比较少。。

**Reply by 0x5459** [reply]
最终确定保留 `附注 (Additional Info)`
